### DISCUSSION TITLE
Is there a way to exclude client_id from form parameters?

### DISCUSSION THREAD

**Discussion by lyndoh** [panva/openid-client#499] [Q&A] [answered]
Hi,

Certain providers, okta being the one I'm currently using, return an error when the client_id is included with the form parameters when performing authentication using a jwt-bearer client assertion, eg. when doing private_key_jwt. For reference, the error returned by okta is like `Cannot supply multiple client credentials. Use one of the following: credentials in the Authorization header, credentials in the post body, or a client_assertion in the post body`.

If I modify the openid-client code to not add the client_id to the form parameters then authentication with okta works ok.

Is there any correct way to exclude the client_id from the form parameters while waiting for okta to fix the issue on their side (if they fix it at all)? If not, I'm happy to provide a PR to do this if required/desired.

Thanks,
Lyndon

**Comment by panva** [comment, marked as answer]
> Is there a way to exclude client_id from form parameters?

There is not. The optionality of body `client_id` is called out in the [underpinning](https://datatracker.ietf.org/doc/html/rfc7521#section-4.2) of `private_key_jwt` and `client_secret_jwt`, it's all within spec (and OIDC certified), a standalone `client_id` in the body is not a "credential" and may always be present regardless of the form of client authentication used.
